src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -158,6 +161,39 @@ class DevTask extends AbstractServerTask {
         this.pollingTest = pollingTest;
     }
 
+    private Boolean container = null;
+
+    @Option(option = 'container', description = 'Run the server in a Docker container instead of locally. The default value is false for the libertyDev task, and true for the libertyDevc task.')
+    void setContainer(boolean container) {
+        this.container = container;
+        project.liberty.dev.container = container; // Needed in DeployTask and AbstractServerTask
+    }
+
+    Boolean getContainer() {
+        return container;
+    }
+
+    private File dockerfile;
+
+    @Option(option = 'dockerfile', description = 'Dev mode will build a docker image from the provided Dockerfile and start a container from the new image.')
+    void setDockerfile(String dockerfile) {
+        if (dockerfile != null) {
+            File temp = new File(dockerfile);
+            try {
+                this.dockerfile = new File(temp.getCanonicalPath()); // ensures the dockerfile is defined with the full path - matches how maven behaves
```

Comment:
I thought setting a dockerfile or dockerRunOpts was supposed to automatically set container to true, but I don't see any code doing that.

Reply:
Fixed by adding a `processContainerParams()` method